accfg/lib/libaccfg.c

```diff
@@ -835,11 +833,7 @@ static void *add_wq(void *parent, int id, const char *wq_base,
 	free(wq_type);
 
 	close(dfd);
-	wq->wq_path = strdup(wq_base);
-	if (!wq->wq_path) {
-		err(ctx, "forming of wq path failed\n");
-		goto err_read;
-	}
+	wq->wq_path = wq_base_string;
```

Comment:
Reason a copy of wq_base was used is because basename() can modify the input path parameter. 
https://man7.org/linux/man-pages/man3/basename.3.html

If we can assume that basename() wouldn't modify the input path then we don't need wq_base_string at all.

Otherwise we can do strcpy(wq_base_string, wq_base) to put back the original path in wq_base.  Maybe add a comment so it would be clear, like "/* basename() could have modified the path */"

Reply:
> Reason a copy of wq_base was used is because basename() can modify the input path parameter. https://man7.org/linux/man-pages/man3/basename.3.html
> 
> If we can assume that basename() wouldn't modify the input path then we don't need wq_base_string at all.
> 
> Otherwise we can do strcpy(wq_base_string, wq_base) to put back the original path in wq_base. Maybe add a comment so it would be clear, like "/* basename() could have modified the path */"

Ugh, yep. Let me fix that up adding those bits back in, and I'll add a comment to make the basename part clear.